man/sd_bus_message_new.xml

```diff
@@ -0,0 +1,171 @@
+<?xml version='1.0'?>
+<!DOCTYPE refentry PUBLIC "-//OASIS//DTD DocBook XML V4.2//EN"
+  "http://www.oasis-open.org/docbook/xml/4.2/docbookx.dtd">
+<!-- SPDX-License-Identifier: LGPL-2.1+ -->
+
+<refentry id="sd_bus_message_new" xmlns:xi="http://www.w3.org/2001/XInclude">
+  <refentryinfo>
+    <title>sd_bus_message_new</title>
+    <productname>systemd</productname>
+  </refentryinfo>
+
+  <refmeta>
+    <refentrytitle>sd_bus_message_new</refentrytitle>
+    <manvolnum>3</manvolnum>
+  </refmeta>
+
+  <refnamediv>
+    <refname>sd_bus_message_new</refname>
+    <refname>sd_bus_message_ref</refname>
+    <refname>sd_bus_message_unref</refname>
+    <refname>sd_bus_message_unrefp</refname>
+    <refname>SD_BUS_MESSAGE_METHOD_CALL</refname>
+    <refname>SD_BUS_MESSAGE_METHOD_RETURN</refname>
+    <refname>SD_BUS_MESSAGE_METHOD_ERROR</refname>
+    <refname>SD_BUS_MESSAGE_SIGNAL</refname>
+
+    <refpurpose>Create a new bus message object and create or destroy references to it</refpurpose>
+  </refnamediv>
+
+  <refsynopsisdiv>
+    <funcsynopsis>
+      <funcsynopsisinfo>#include &lt;systemd/sd-bus.h&gt;</funcsynopsisinfo>
+
+      <funcsynopsisinfo><token>enum</token> {
+      <constant>SD_BUS_MESSAGE_METHOD_CALL</constant>,
+      <constant>SD_BUS_MESSAGE_METHOD_RETURN</constant>,
+      <constant>SD_BUS_MESSAGE_METHOD_ERROR</constant>,
+      <constant>SD_BUS_MESSAGE_SIGNAL</constant>,
+};</funcsynopsisinfo>
+
+      <funcprototype>
+        <funcdef>int <function>sd_bus_message_new</function></funcdef>
+        <paramdef>sd_bus *<parameter>bus</parameter></paramdef>
+        <paramdef>sd_bus_message **<parameter>m</parameter></paramdef>
+        <paramdef>uint8_t <parameter>type</parameter></paramdef>
+      </funcprototype>
+
+      <funcprototype>
+        <funcdef>sd_bus_message *<function>sd_bus_message_ref</function></funcdef>
+        <paramdef>sd_bus_message *<parameter>m</parameter></paramdef>
+      </funcprototype>
+
+      <funcprototype>
+        <funcdef>sd_bus_message *<function>sd_bus_message_unref</function></funcdef>
+        <paramdef>sd_bus_message *<parameter>m</parameter></paramdef>
+      </funcprototype>
+
+      <funcprototype>
+        <funcdef>void <function>sd_bus_message_unrefp</function></funcdef>
+        <paramdef>sd_bus_message **<parameter>mp</parameter></paramdef>
+      </funcprototype>
+    </funcsynopsis>
+  </refsynopsisdiv>
+
+  <refsect1>
+    <title>Description</title>
+
+    <para><function>sd_bus_message_new()</function> creates a new bus message object attached to the
+    bus <parameter>bus</parameter> and returns it in the output parameter <parameter>m</parameter>.
+    This object is reference-counted, and will be destroyed when all references are gone. Initially,
+    the caller of this function owns the sole reference to the message object. Note that the message
+    object holds a reference to the bus object, so the bus object will not be destroyed as long as
+    the message exists.</para>
```

Comment:
maybe mention that sd_bus_message_new() is low-level, and people should use sd_bus_message_new_signal() and sd_bus_message_new_method_call() and such instead, unless they know what they do